os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -192,6 +198,23 @@ int elf_addrenv_alloc(FAR struct elf_loadinfo_s *loadinfo, size_t textsize, size
 	loadinfo->binp->uheap_size = loadinfo->binp->ramsize - (loadinfo->textsize + loadinfo->datasize) - sizeof(struct mm_heap_s);
 #endif
 	loadinfo->binp->heapstart = loadinfo->dataalloc + loadinfo->datasize;
+
+#ifdef CONFIG_DEBUG_BIN_SYMBOL
+	bin_addr_info_t *bin_info;
+	bin_info = (bin_addr_info_t *)kmm_malloc(sizeof(bin_addr_info_t));
+	if (bin_info != NULL) {
+		strncpy(bin_info->bin_name, loadinfo->binp->bin_name, BIN_NAME_MAX);
+		bin_info->text_addr = loadinfo->textalloc;
+		bin_info->rodata_addr = loadinfo->roalloc;
+		bin_info->data_addr = loadinfo->dataalloc;
+		bin_info->bss_addr = loadinfo->binp->bssstart;
+
+		sq_addlast((sq_entry_t *)bin_info, &g_bin_addr_list);
+	} else {
+		berr("ERROR : Failed to allocate for bin symbol.\n");
+	}
+#endif
+
```

Comment:
How about moving this code to  binfmt_loadbinary.c ?
If there is an error in loading binary after this point, then binary will not be loaded, but the bin_info will remain in the list. So, better to create the bin_info only upon successful loading of binary.